forum/qa-plugin/ckeditor4/qa-ckeditor4-layer.php

```diff
@@ -60,13 +57,44 @@ function main_part($key, $part)
                     $html .= '</span>';
 
                     $content = &$part['fields']['content'];
-                    if (isset($content['label']))
-                        $content['label'] = $html . $content['label'];
-                    else
-                        $content['label'] = $html;
+                    $content['label'] = $html;
+                    if (isset($content['label'])) {
+                        $content['label'] .= $content['label'];
+                    }
                 }
             }
         }
-        qa_html_theme_base::main_part($key, $part);
+
+        parent::main_part($key, $part);
+    }
+
+    public function head_css()
+    {
+        $this->content['css_src'][] = '/qa-content/css/shCore.css?v=' . QA_RESOURCE_VERSION;
+        $this->content['css_src'][] = '/qa-content/css/shThemeDefault.css?v=' . QA_RESOURCE_VERSION;
+
+        parent::head_css();
+    }
+
+    public function main()
+    {
+        parent::main();
+
+        $scriptsOutput = '';
+
+        $scripts = [
+            'shCore', 'shLegacy', 'shBrushBash', 'shBrushCpp', 'shBrushCSharp', 'shBrushCss', 'shBrushDelphi',
+            'shBrushJava', 'shBrushJScript', 'shBrushPerl', 'shBrushPhp', 'shBrushPlain', 'shBrushPowerShell',
+            'shBrushPython', 'shBrushRuby', 'shBrushSql', 'shBrushVb', 'shBrushXml'
+        ];
+        foreach ($scripts as $script) {
+            $path = "/qa-content/javascript/{$script}.js?v=" . QA_RESOURCE_VERSION;
+            $scriptsOutput .= '<script src="' . $path . '"></script>';

```

Comment:
Dało by się to przenieść do <head>, żeby skrypty zaczytywały się wcześniej + ustawić `defer` żeby ładować je asynchronicznie?

Reply:
Ogarnąłem to.